### PR TITLE
Feature/local db lookup

### PR DESCRIPTION
**Pull request by csucu**
So the lookup function for new urls, first checks the local db, then queries  googles. We wanted away todo the local search separately, leaving it up-to the app if we should do the second part or not.

Added 2 changes:

1. First is to modify the search endpoint in the server to accept multiple urls in one request(this is just for efficiency, as we will be sending loads of urls, client accepts multiple at once anyway)
2. Second is to add the flag `local` to server request and client lookup argument to specify if we should only search the local db.

### CODE REVIEW

**Review by dmotylev — COMMENTED** [review]
So, there are two changes:
- `local` flag
- `repeated` option added to URI fields in some messages

Looks ok from a briefly scan. Do we need to replace google with redsift? There is a licence and copyright over all the code; we might use go modules `replace` feature for hosting our changes in the fork and still use the package as `github.com/google/...`

**Comment by csucu** [comment]
> So, there are two changes:
> 
> * `local` flag
> * `repeated` option added to URI fields in some messages
> 
> Looks ok from a briefly scan. Do we need to replace google with redsift? There is a licence and copyright over all the code; we might use go modules `replace` feature for hosting our changes in the fork and still use the package as `github.com/google/...`

Could do, but replace only works with go modules, so will have to turn it modules project

File: internal/webrisk_proto/webrisk.proto [hunk 1]
```diff
@@ -100,25 +102,30 @@ message ComputeThreatListDiffResponse {
 // Request to check URI entries against threatLists.
 message SearchUrisRequest {
   // The URI to be checked for matches.
-  string uri = 1;
+  repeated string uris = 1;
```
Comment (jpicht) [inline internal/webrisk_proto/webrisk.proto:105]: Won't it break backwards compatibility to change a type?

File: internal/webrisk_proto/webrisk.proto [hunk 2]
```diff
@@ -100,25 +102,30 @@ message ComputeThreatListDiffResponse {
...
   // Required. The ThreatLists to search in.
   repeated ThreatType threat_types = 2;
+
+  bool local = 3;
 }
 
 message SearchUrisResponse {
   // Contains threat information on a matching uri.
   message ThreatUri {
+    // The pattern for the URI
```
Comment (jpicht) [inline internal/webrisk_proto/webrisk.proto:116]: Inserting at the front is breaking backwards compatibility.

**Review by jpicht — COMMENTED** [review]
I don't know about the consumers of the PBF API, but it's generally considered bad practice to break backwards compatibility.

File: internal/webrisk_proto/webrisk.proto (hunk 1 above)
Comment (csucu) [reply internal/webrisk_proto/webrisk.proto:105]: Didn't think backwards compatibility would be an issue. We only use this is ipforensics, and wasn't planning to merge this back into upstream anyway. I don't think they really even maintain that library anymore.

File: internal/webrisk_proto/webrisk.proto (hunk 2 above)
Comment (csucu) [reply internal/webrisk_proto/webrisk.proto:116]: Read comment above for backwards compatibility

**Review by jpicht — APPROVED** [review]

**Pull request merged**